
mawk pattern matching is probably too greedy as it fails when two paths on same line:


mawk: run time error: regular expression compile failed (missing operand)
/usr/lib/nullmailer' || -d '/usr/local/lib/nullmailer
        FILENAME="check_mailq.pl" FNR=627 NR=627


this is resolved differently in monitoring-plugins:
https://github.com/monitoring-plugins/monitoring-plugins/commit/c08d6a429ba0e0cd3642ba2c2fe85687472ee22f

--- nagios-plugins-2.0.1/plugins-scripts/check_mailq.pl~	2014-04-21 14:33:35.000000000 +0300
+++ nagios-plugins-2.0.1/plugins-scripts/check_mailq.pl	2014-04-21 14:33:37.760719061 +0300
@@ -610,17 +610,22 @@
 		{
 			$mailq = 'qmail';
 		}
-		elsif (-d '/var/lib/postfix' || -d '/var/local/lib/postfix'
-		       || -e '/usr/sbin/postfix' || -e '/usr/local/sbin/postfix')
+		elsif (-d '/var/lib/postfix'
+			|| -d '/var/local/lib/postfix'
+		       || -e '/usr/sbin/postfix'
+			   || -e '/usr/local/sbin/postfix')
 		{
 			$mailq = 'postfix';
 		}
-		elsif (-d '/usr/lib/exim4' || -d '/usr/local/lib/exim4'
-		       || -e '/usr/sbin/exim' || -e '/usr/local/sbin/exim')
+		elsif (-d '/usr/lib/exim4'
+			|| -d '/usr/local/lib/exim4'
+		       || -e '/usr/sbin/exim'
+			   || -e '/usr/local/sbin/exim')
 		{
 			$mailq = 'exim';
 		}
-		elsif (-d '/usr/lib/nullmailer' || -d '/usr/local/lib/nullmailer'
+		elsif (-d '/usr/lib/nullmailer'
+			|| -d '/usr/local/lib/nullmailer'
 		       || -e '/usr/sbin/nullmailer-send'
 		       || -e '/usr/local/sbin/nullmailer-send')
 		{
